In [1]:
import numpy as np
import seaborn as sns
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
data = pd.read_csv("cleaned_hotel_bookings.csv")

# Felnőttek, gyerekek, babák

Következőnek ezekkel a feature-kat szeretném megvizsgálni. Az én gondolatom a következő. Minél több személy foglalta le a szállást annál, nagyobb a valószínűsége, hogy a szállás le lesz mondva, hiszen minél több ember tervezz előre annál valószínűbb, hogy az egyikkel történi fog valami, lebetegszik, mégis csak be kell mennie a munkába, hasonlóan a gyerekkel.

### A következő állításokat szeretném megvizsgálni:

- Minél több adults van a foglalásban annál valószínűbb, hogy lemondják
- Ha van legalább 1 children a foglalásban, nagyobb a valószínűsége, hogy lemondják mintha nem lenne
- Ha van legalább 1 babies a foglalásban, nagyobb a valószínűsége, hogy lemondják mintha nem lenne
- Minél több a személyek(adults, children, babies) száma összesen, annál valószínűbb, hogy lemondják

### Minél több adults van a foglalásban annál valószínűbb, hogy lemondják vizsgálata

Először megnézem mennyi az átlagos adults az összes foglalásban

In [3]:
print("Átlagos felnőttek száma az összes foglalás figyelembe vételénél: {:.4f}".format(data["adults"].mean()))

Átlagos felnőttek száma az összes foglalás figyelembe vételénél: 1.8564


Külön megnézve az átlagos felnőttek száma, a lemondott és a megérkezett foglalásoknál

In [4]:
pd.pivot_table(data, values = "adults", index = "is_canceled")

adults
is_canceled          
0            1.829737
1            1.901728

A lemondott foglalásoknál valamivel több az átlagos felnőttek száma azonban a két érték nem tér el nagyban egymástól, hogy egyértelmű következtetést tudjunk levonni belőle

### Ha van legalább 1 children a foglalásban, nagyobb a valószínűsége, hogy lemondják mintha nem lenne vizsgálata

In [36]:
no_children_all = data.loc[data["children"] == 0, "is_canceled"].count()
children_all = data.loc[data["children"] != 0, "is_canceled"].count()
print(f"{no_children_all} db foglalást children nélkül foglaltak")
print(f"{children_all} db foglalást legalább 1 children-nel foglaltak")
no_children_canceled = data.loc[(data["children"] == 0) & (data["is_canceled"] == 1), "is_canceled"].count()
children_canceled = data.loc[(data["children"] != 0) & (data["is_canceled"] == 1), "is_canceled"].count()
print("Nincs children a foglalásban lemondva: {:.4f}".format(no_children_canceled/no_children_all))
print("Legalább 1 children van a foglalásban lemondva: {:.4f}".format(children_canceled/children_all))

110800 db foglalást children nélkül foglaltak
8590 db foglalást legalább 1 children-nel foglaltak
Nincs children a foglalásban lemondva: 0.3709
Legalább 1 children van a foglalásban lemondva: 0.3639


A két eset között, hogy legalább 1 children-nel vagy children nélkül mondják-e le nagyobb valószínűséggel a foglalást nincs nagy eltérés. Valamint a két különböző esetnek nagyban eltér a rendelkezésre álló adatmennyiség (egyik esetről 110800 példánk van a másikról csupán 8590), ezért az állítást, hogy akik legalább 1 children-nel foglaltak szállást nagyobb valószínűséggel mondják nem tudom igazolni.

### Ha van legalább 1 babies a foglalásban, nagyobb a valószínűsége, hogy lemondják mintha nem lenne

In [37]:
no_babies_all = data.loc[data["babies"] == 0, "is_canceled"].count()
babies_all = data.loc[data["babies"] != 0, "is_canceled"].count()
no_babies_canceled = data.loc[(data["babies"] == 0) & (data["is_canceled"] == 1), "is_canceled"].count()
babies_canceled = data.loc[(data["babies"] != 0) & (data["is_canceled"] == 1), "is_canceled"].count()
print(f"{no_babies_all} db foglalást babies nélkül foglaltak")
print(f"{babies_all} db foglalást legalább 1 babies-el foglaltak")
print("Nincs babies a foglalásban lemondva: {:.4f}".format(no_babies_canceled/no_babies_all))
print("Legalább 1 babies van a foglalásban lemondva: {:.4f}".format(babies_canceled/babies_all))

118473 db foglalást babies nélkül foglaltak
917 db foglalást legalább 1 babies-el foglaltak
Nincs babies a foglalásban lemondva: 0.3719
Legalább 1 babies van a foglalásban lemondva: 0.1821


Ebben a két esetben már elégnek mondható lenne az eltérés, hogy következtetéseket vonjunk le. Azonban a rendelkezésre álló adatok arányával nagyobb baj van mint az előző vizsgálatnál, hiszen itt azoknak a foglalásoknak a száma amiben legalább 1 babies van az 1000 darabot sem éri el míg a másik esetben 118000-nél is több példánk van. Emiatt a hatalmas aránytalanság miatt ebből sem tudunk biztos dolgot állítani.

### Minél több a személyek(adults, children, babies) száma összesen, annál valószínűbb, hogy lemondják vizsgálata

In [40]:
data["total_people"] = data["adults"]+data["children"]+data["babies"]
pd.pivot_table(data, values = "total_people", index = "is_canceled")

total_people
is_canceled              
0                1.942461
1                2.012052

A két érték ebben az esetben sem tér el egymástól nagyon, így megint nem tudunk biztos következtetéseket levonni

Szakdogában arról írni ezután a rész után, hogy miért lehetett az hogy még a 3_day_night_weekday_weekend_months munkafüzetben volt hasznos következtetésünk amit bizonyitani is tudtunk, ebben a részben azonban már egyáltalán nincsen ilyen. Korrelációs mátrix bevezetése stb.